### ISSUE TITLE
[Bug]: Can't obviously control radial ticks on polar plot

### ISSUE THREAD

**Issue by robbiemallett** [matplotlib/matplotlib#22104]
### Bug summary

The fontsize argument of ax.set_xticklabels() controls the size of azimuthal tick labels on a polar plot.
However the same argument, when passed to set_yticklabels(), does not control the size of radial tick labels. In fact it does nothing - the value of the argument has no impact on the plot.
To modify the size of radial tick labels, one must call: ax.yaxis.set_tick_params(labelsize='large')
This is not obvious.

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np

# Make fake data

d = np.random.rand(11,18)
azimuths = np.radians(np.arange(-42.5,42.6,5))
radials = np.linspace(1.3,3.9,11)
r, theta = np.meshgrid(radials, azimuths)

# Plot the data

fig = plt.figure(figsize=(6,6))

ax1 = fig.add_subplot(111,polar=True)

cont = ax1.contourf(theta, r, np.fliplr(d).T,
                   )

ax1.set_theta_zero_location('N')
ax1.set_thetamin(-45)
ax1.set_thetamax(45)
ax1.set_xticks(np.arange(np.radians(-45),
                        np.radians(46),
                        np.radians(15),
                       ))  # Less radial ticks


ax1.set_xticklabels(np.arange(-45,46,15)[::-1],
                    fontsize='xx-large',            # This argument controls azimuthal tick labels
                   )

ax1.set_yticklabels(np.arange(-45,46,15)[::-1],
                    fontsize='xx-large',            # So why does this not control the radial tick labels?
                   )

# ax1.yaxis.set_tick_params(labelsize='xx-large')      # Especially when this line *does* control them

plt.show()
```

### Actual outcome

Here you can see that the radial tick labels are too small, unaffected by the set_yticklabels command:

![download](https://user-images.githubusercontent.com/47150532/148107999-89de7288-5491-4690-998e-8b4b90dbfe48.png)

### Expected outcome

They should be this size. The following is produced by using the set_tick_params command:

![download](https://user-images.githubusercontent.com/47150532/148108446-ccfaed14-3d49-475b-93f2-0971fd5878d5.png)

### Operating system

Ubuntu 18.04

### Matplotlib Version

This occurs with matplotlib v3.3.4

### Matplotlib Backend

inline

### Python version

3.7.10

### Jupyter version

6.4.6

### Installation

conda

**Comment by lxguan1** [comment]
Hi,

I plan on doing a pr for a solution to this issue, but have you found any other examples of this bug occurring?

It seems to be that set_yticklabels does not save the fontweight setting, so when a new tick is created later it does not have the fontweight applied.

Thanks!

**Comment by timhoffm** [comment]
Without checking (I’m currently away from my dev machine) I suspect that set*ticklabels kwargs only sets the properties of the tick labels defined therein, but not the general tick label properties. - We should discuss whether that’s desired or not.

I also assume this is only indirectly  related to polar. The reason for the difference between r and theta here is likely that r limits are not fixed and data limits cause auto-scaling of the r-range.

**Comment by lxguan1** [comment]
@timhoffm Yes, I have looked into this and that seems to be the case. set_yticklabels is only setting the properties of the tick labels passed in, and with polar axes confined to a sector (the example given in the issue) reset_ticks() is called in the draw() function in polar.py (line 958), so the ticks that are created do not have the fontize property set. My current thoughts on fixing this is to call set_tick_params in set_ticklabels (you can see the changes I did in my commit), but it seems that it is a significant behavior change.

**Comment by timhoffm** [comment]
> My current thoughts on fixing this is to call set_tick_params in set_ticklabels (you can see the changes I did in my commit), but it seems that it is a significant behavior change.

This is indeed a major behavior change, which should be agreed upon by core devs. However, I'm inclinded towards this because I think the sematics of styling in `set_ticklabels()` is global (not only universal for all the ticks set, but also for future ticks). The major problem I see is that `set_tick_params` only covers a small portion of Text properties, that one could pass to `set_ticklabels()` ( from a quick look possibly: `labelcolor`, `labelrotation`, `labelsize`, `zorder`, and the latter cannot be set independently of the tick line). And different behavior for different parameters would be even worse.

Possibly the first step forward is improving the `set_ticklabel` and `set_ticks` docs to state the current behavior that it applies only to the given ticks, and if possible users should use `tick_params` instead. Everything behavior change is probably a larger effort.

**Comment by rcomer** [comment]
`set_ticklablels` was formerly discouraged at #20047, and the code from the OP now throws a reasonably clear warning:

```
UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.
```

So I do not think there is anything more to do here.  Please comment and/or re-open if you disagree.
